core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -674,7 +675,10 @@ bool core_mmu_divide_block(struct core_mmu_table_info *tbl_info,
 		return false;
 
 	entry = (uint64_t *)tbl_info->table + idx;
-	assert((*entry & DESC_ENTRY_TYPE_MASK) == BLOCK_DESC);
+	assert(!*entry || (*entry & DESC_ENTRY_TYPE_MASK) == BLOCK_DESC);
+
+	/* We need to flush TLBs only if there already was some mapping */
+	flush_tlb = *entry;
```

Comment:
Aha, interesting, thanks.
To see it in action I disassembled the code (compiled with -O0) and there's:
```
  50:   2b00            cmp     r3, #0
  52:   bf14            ite     ne
  54:   2301            movne   r3, #1
  56:   2300            moveq   r3, #0
  58:   74fb            strb    r3, [r7, #19]
```